pkg/apis/migration/v1alpha1/gvk.go

```diff
@@ -44,3 +50,72 @@ func (i *Incompatible) ResourceList() (incompatible []string) {
 	}
 	return
 }
+
+func CollectResources(discovery discovery.DiscoveryInterface) ([]*metav1.APIResourceList, error) {
```

Comment:
Agreed they are generic GVK functionality.  Why in the API package?

Reply:
May be it is not needed, the only problem that I hit a dependency loop at some point, trying to use an internal `gvk` implementation. If the GVK was initially dedicated to a separate package, this won't happen, but right now it is necessary evil.

Reply:
@jortel does it sound like a solution to you?